canonicalwebteam/discourse_docs/blueprint.py

```diff
@@ -0,0 +1,53 @@
+import flask
+from requests.exceptions import HTTPError
+
+from canonicalwebteam.discourse_docs.models import (
+    NavigationParseError,
+    RedirectFoundError,
+)
+
+
+def build_blueprint(url_prefix, model):
+
+    blueprint = flask.Blueprint("discourse_docs", __name__)
+
+    def redirect_handler(path):
+        if url_prefix != "/":
+            return flask.redirect(url_prefix + path)
+        else:
+            return flask.redirect(path)
+
+    @blueprint.route("/")
+    def homepage():
+        """
+        Redirect to the frontpage topic
+        """
+        frontpage, nav_html = model.parse_frontpage()
+        return redirect_handler(frontpage["path"])
+
+    @blueprint.route("/t/<path:path>")
+    def redirect(path):
+        return redirect_handler("/" + path)
+
+    @blueprint.route("/<path:path>")
+    def document(path):
+        try:
+            document, nav_html = model.get_document(path)
+        except RedirectFoundError as redirect_error:
+            return flask.redirect(redirect_error.redirect_path)
+        except HTTPError as http_error:
+            flask.abort(http_error.response.status_code)
+        except NavigationParseError as nav_error:
+            document = nav_error.document
+            nav_html = f"<p>{str(nav_error)}</p>"
+
+        return flask.render_template(
+            "docs/document.html",
```

Comment:
This doesn't *need* to be done in this PR, but:

For the very first use-case for this, docs.snapcraft.io, the template actually sits at `templates/document.html`. Which makes sense, given the whole site is about docs.

So right off the bat, we're going to have a use-case where we want to overwrite this path. So should we make it customisable right away in this PR? `def build_blueprint(model, url_prefix="docs", document_template="docs/document.html"):` ?

Reply:
done